File: Systeme-Gestion-Stock-abs-front/src/app/models/detail-sortie.ts
import { Produit } from './produit';
import { BonSortie } from './bon-sortie';
export interface DetailSortie {
  details_sortie_id: number;
  quantity: number;
  prix_total: number;
  date_expiration: Date;
  bon_sortie_id: number;
  produit_id: number;
}

File: Test-SB/src/main/java/com/test/entities/DetailSortie.java
package com.test.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Table(name = "details_sorties")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class DetailSortie {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int Id;

    @Column(name = "quantity")
    private int quantity;

    @Column(name = "prix_total")
    private int prixTotal;

    @Column(name = "date_expiration")
    private LocalDate dateExpiration;

    @ManyToOne
    @JoinColumn(name = "bon_sortie_id", nullable = false)
    private BonSortie bonSortie;

    @ManyToOne
    @JoinColumn(name = "produit_id", nullable = false)
    private Produit produit;
}
